src/widgets/ot_carousel.eliom

```diff
@@ -488,6 +487,24 @@ let%shared make
   in
   {elt = d; pos = pos_signal; vis_elts = nb_visible_elements; swipe_pos}
 
+let%shared spinner () = D.div ~a:[a_class ["ot-icon-animation-spinning"]] []
+
+(* on the client side we generate the contents of the initially visible page
+   asynchronously so the tabs will be rendered right away *)
+let%client generate_initial_contents gen =
+  let s = spinner () in
+  begin Lwt.async @@ fun () ->
+    let%lwt contents = Eliom_shared.Value.local gen () in
+    Manip.replaceSelf s contents;
```

Comment:
This is not going to work if the contents is generated at once: indeed, the spinner does not have a parent yet at this point.

Thus, `Manip.replaceSelf` should somehow be called after the call to `make` below. Also, one should call `To_dom.of_element carousel.elt` before that, to be sure that the DOM elements corresponding to the carousel have been created.